### ISSUE TITLE
Projects detail view

### ISSUE THREAD

**Issue by javieralaves** [AutarkLabs/product#47]
## Description

Enable a Projects detail view that directs users to a detail view with a list of issues for that repo, instead of having a single tab called `Issues` where cross-repo issues are grouped.

### Acceptance criteria

- [ ] The 'Issues' tab is no longer visible in the `Tabs` area
- [ ] Clicking on a project takes you to its detail view
- [ ] Issues in a project detail view can be filtered based on the rules defined in the spec
- [ ] Issues in a project detail view can be searched
  - [ ] A 'Not found' state is triggered when no results are found
- [ ] Issues in a project detail view can be sorted by creation date, alphabetically, by value and by bounty phase
- [ ] 'New issue' is the global button in project detail view, and once clicked, it will be pre-populated to the project from which the button was clicked

### Specifications

[Available in Google Drive.](https://docs.google.com/document/d/185wMhoDQ4h1G51MLkRzLtqYDdybm3JzRlYCvPR0iC5Y/edit?usp=sharing)

**Comment by javieralaves** [comment]
@stellarmagnet Epic specifications are ready for your product review.

**Comment by stellarmagnet** [comment]
@javieralaves I made a couple comments to the spec about the filtering behavior.

One thing I would edit for the acceptance criteria is change this:
> Issues in a project detail view can be filtered by Project and Status

to
> Issues in a project detail view can be filtered based on the rules defined in the spec

Since there are quite a few different things to check for

**Comment by stellarmagnet** [comment]
Acceptance criteria wording tweak. Before:
> Global button in project view is set to 'New issue'
> Project dropdown pre populates to the project from which the button was clicked on

Reword:

> 'New issue' is the global button in project detail view, and once clicked, it will be pre-populated to the project from which the button was clicked

Perhaps next time, the acceptance criteria can be included in the Google doc, and once the spec is finalized can be placed into the epic

**Comment by javieralaves** [comment]
@stellarmagnet Thanks Yalda, I've reflected the changes; and you're right. We'll have acceptance criteria be part of the Google Doc specifications before adding it to GitHub for editing purposes. I think this epic is nearly ready to go, please have a look at my comments re: filtering in the Google Doc.

**Comment by stellarmagnet** [comment]
@javieralaves it doesn't look like all of my comments in the spec were addressed - bouncing this back to you!

**Comment by stellarmagnet** [comment]
Let's move this to technical review so the dev team can start reviewing it. The remaining comments seem like they can be addressed in parallel or they require input from devs.

**Comment by chadoh** [comment]
If we want to limit this epic as much as possible and ensure that we can get it done quickly, I would like to treat it only as "rename Issues tab to Project Detail". If that's how we think about this, then acceptance criteria still seem relevant to me:

- The 'Issues' tab is no longer visible in the Tabs area
- Clicking on a project takes you to its detail view

But the rest seem like they should essentially be "keep it working as much like current Issues tab as is sensible." Maybe these would be the other acceptance criteria:

- "Projects" dropdown becomes "Project" dropdown and is now a single-select field _[OR: Page title changes from "Projects" to the name of the currently selected project and "Projects" dropdown is removed]_
- Filter bar shows back button, which returns to the General tab

Oh, and I think we need one for:

- Rename "Overview" tab to "General"

All of the others, though, I think we should get rid of. We can keep other searching/filtering/sorting functionality all unchanged.

@javieralaves @stellarmagnet what do y'all think?

**Comment by chadoh** [comment]
I would also be ok with including filtering improvements into this epic – I guess it's the name of the epic that has me wondering if we care about doing any improvements right now.

**Comment by javieralaves** [comment]
@chadoh Sounds good to me. The focus is certainly not on improved filters; let's consider it a very nice to have that has made sense to include in this update all together. It won't hurt if we don't get to those improvements in this sprint because we need to deal with higher priority items.

**Comment by javieralaves** [comment]
@stellarmagnet Chad suggested that we render global header titles conditionally based on a) whether the user is within a project detail view to display the project name as header instead of 'Projects' and b) whether the user is within a issue detail view to display the issue name as header instead of 'Projects'. 

I think that for projects it works well because that way we can get rid of the 'Project' filter in the filter area. For issues it may not work so well right now because there's redundancy given that we're already displaying the issue name right below. 

So overall, I'd suggest we move forward with a). Safe enough to try?
